src/rgw/rgw_rest_log.h

```diff
@@ -275,6 +276,21 @@ class RGWOp_DATALog_Notify : public RGWRESTOp {
   }
 };
 
+class RGWOp_DATALog_Notify2 : public RGWRESTOp {
+  rgw_data_notify_entry data_notify;
+public:
+  RGWOp_DATALog_Notify2() {}
+  ~RGWOp_DATALog_Notify2() override {}
+
+  int check_caps(const RGWUserCaps& caps) override {
+    return caps.check_cap("datalog", RGW_CAP_WRITE);
+  }
+  void execute(optional_yield y) override;
+  const char* name() const override {
+    return "datalog_notify2";
+  }
+};
+
```

Comment:
this class doesn't need to be in the header, since `RGWHandler_Log::op_post()` is the only place we depend on its declaration. you can move it into rgw_rest_log.cc just above `void RGWOp_DATALog_Notify2::execute()`. that can save us the dependency on "rgw_datalog.h" here

Reply:
looks like class RGWOp_DATALog_ShardInfo already has a dependency on rgw_datalog.h.